workflows/validate.py

```diff
@@ -0,0 +1,78 @@
+"""
+Runs the json validator on a json file and raises an exception if there is an error
+Intended to be used as a hook for repositories which are developing recipes
+
+Example of how this could be used in a .pre-commit-config.yaml file:
+```
+# Validate json recipes
+- repo: https://github.com/DiamondLightSource/python-workflows
+  hooks:
+  - name: Recipe validation
+    id: recipe-validation
+    language: python
+    files: ^recipes/
+    entry: workflows-recipe-validate
+```
+"""
+
+import logging
+import workflows.recipe
+import workflows
+import sys
+import json
+import argparse
+
+
+def validate_recipe(json_filename):
+    """Reads a json file, tries to turn it into a recipe and then validates it.
+    Exits on exception with non-zero error"""
+
+    # Read in the file
+    try:
+        with open(json_filename) as f:
+            recipe_text = f.read()
+    except Exception as e:
+        logging.exception("Could not read recipe from {0}".format(json_filename))
+        raise e
```

Comment:
If you're not doing anything with `e` then you don't need it in scope. `except Exception: raise` does this automatically.

Reply:
Done